src/main/java/gillycarpetaddons/helpers/EndPortalFrameHelper.java

```diff
@@ -0,0 +1,69 @@
+package gillycarpetaddons.helpers;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.EndPortalFrameBlock;
+import net.minecraft.item.EnderEyeItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.ItemUsageContext;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public final class EndPortalFrameHelper {
+    /**
+     * Checks if the block state has an eye of ender
+     * @param blockState the block state to check
+     * @author Gilly7CE
+     */
+    public static boolean HasEyeOfEnder(BlockState blockState) {
+        return blockState.get(EndPortalFrameBlock.EYE);
+    }
+
+    /**
+     * Sets an end portal frame blocks state to full
+     * @param item the ender eye item
+     * @param context the item usage context
+     * @author Gilly7CE
+     */
+    public static void setFullEndPortalFrameState(EnderEyeItem item,
+                                                  ItemUsageContext context) {
+        item.useOnBlock(context);
+    }
+
+    /**
+     * Sets an end portal frame blocks state to empty and breaks any connecting portals
+     * @param world the current world
+     * @param blockState the current block state
+     * @param pos the current block position
+     * @author serz
+     */
+    public static void setEmptyEndPortalFrameState(World world, BlockState blockState, BlockPos pos) {
+        BlockState newState = blockState.with(EndPortalFrameBlock.EYE, false);
+        world.setBlockState(pos, newState);
+        //remove portal this frame used to create
+        removePortal(world, pos.north());
+        removePortal(world, pos.west());
+        removePortal(world, pos.south());
+        removePortal(world, pos.east());
```

Comment:
You can make it faster and more accurate by replacing quoted code with:
removePortal(world, pos.offset(state.get(FACING),1));

Reply:
Good suggestion! Refactored in latest changes.